### PR TITLE
Add maximum traversals threshold to AStarGrid2D

### PR DESCRIPTION
**Pull request by Veradictus**
Provides support for exhausting search function after searching a certain number of nodes. This is extremely useful for large tile-map applications where the player attempts to click on a blocked off region and we do not want to search the entirety of the map, thus blocking the thread. `-1` disables this functionality altogether, recommended value is ~`600` depending on how far you are zoomed out.

### CODE REVIEW

File: core/math/a_star_grid_2d.cpp [hunk 1]
```diff
@@ -441,6 +450,11 @@ bool AStarGrid2D::_solve(Point *p_begin_point, Point *p_end_point) {
 			break;
 		}
 
+		// If we have a limit on traversals, increment and stop once we reach the threshold.
+		if (max_traversals != -1 && traversal_count++ >= max_traversals) {
```
Comment (AThousandShips) [inline core/math/a_star_grid_2d.cpp:454]: We like to avoid incrementing in a check because it makes it hard to read, I'd suggest breaking this up

Comment (Veradictus) [reply core/math/a_star_grid_2d.cpp]: Sounds good, fixed.

**Comment by Veradictus** [comment]
> Does a value of `0` make sense? If not, I suggest making `0` count as "unlimited" and make it the default value.

Fixed

**Comment by AThousandShips** [comment]
You've included a lot of unrelated files, please remove them

**Comment by AThousandShips** [comment]
You are using your `master` branch for this PR meaning all changes you do on that branch gets included, I'd suggest starting over on a new branch, or removing any unrelated code and putting it in a separate branch until you're done with this

**Comment by lapspider45** [comment]
Looks like the folks at Redot managed to clean this PR up: https://github.com/Redot-Engine/redot-engine/pull/751/commits/613fa6ee29d96f1732959a8ce128b35e3d48da00
Maybe it could get merged into Godot now as well?

**Pull request closed without merging**